compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/scene/SwingComposeSceneLayer.desktop.kt

```diff
@@ -52,8 +55,8 @@ internal class SwingComposeSceneLayer(
         override fun addNotify() {
             super.addNotify()
             mediator?.onComponentAttached()
-            _boundsInWindow?.let {
-                mediator?.contentComponent?.bounds = it.toAwtRectangle(density)
+            if (!visibleBounds.isEmpty) {
```

Comment:
Why only if it's not empty? What if something big was drawn and then nothing was drawn?

Reply:
> Why only if it's not empty?

It's an additional check on `addNotify` as an early attempt to set real bounds if available. Initially it set as the main layer's bounds to allow it to actually be drawn

> What if something big was drawn and then nothing was drawn?

Any changes will be handled in `onDrawRectChange`